remoulade/state/backends/redis.py

```diff
@@ -38,7 +38,7 @@ def get_state(self, message_id):
     def set_state(self, state, ttl=3600):
         message_key = self._build_message_key(state.message_id)
         with self.client.pipeline() as pipe:
-            encoded_state = self._encode_dict(state.as_dict())
+            encoded_state = self._encode_dict(state.as_dict(encode_args=True))
```

Comment:
I don't think you should use `encode_args=True` here, only for retrun of route functions.